### PR TITLE
Fix territory shortcuts when census tract is selected

### LINKED ISSUE #1056: Shortcuts broken after tract selection
**Describe the bug**
When I zoom in anywhere and select a tract to view, and THEN click the "AS" shortcut, the map moves to the left slightly instead of taking me to American Samoa.

**To Reproduce**
1. Go to the Explore the Tool page.
2. Zoom into any census tract NOT near American Samoa. 
3. Click the tract.
4. Click the AS shortcut
5. See that the map only moves slightly to the left.

**Expected behavior**
I expected the map to immediately jump to and zoom in on American Samoa. If you click the AS shortcut multiple times, it finally will move you to American Samoa, but it should only need one click.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MacOS Mojave
 - Browser [e.g. chrome, safari]- Firefox
 - Version [e.g. 22] - 89

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by vim-usds**
# Problem
It seems that when a tract is selected and/or the zoom level was above 8 (z>8), when clicking on a _territory shortcut_, e.g., `AS`, it would move the map over a tiny bit. This is because the click was registering as a click on the _map_ and NOT the territory shortcut `AS`.

### Solution (TL;DR)
We had two click handlers (one for _territory shortcuts_ and one for the _map_) firing each time the map was clicked. This meant either handler could handle the event. Normally `preventDefault()` or `stopPropagation()` are used to solve this, however for some reason it was not working in this case. 

Now we have a single click handler (aka [Event Delegation Pattern](https://dmitripavlutin.com/javascript-event-delegation/)) to handle both the _territory shortcut_ clicks and _map_ clicks. Now we decide based on the event's payload if it's a _map_ event or a _territory shortcut_ event.  

# Purpose of PR

- Re-architects the territory controls which was done by Nat. I believe it's important that we exhaustively test this. While I was hoping this would have fixed the automated tests for this [existing issue](https://github.com/usds/justice40-tool/issues/579), unfortunately it doesn't and so we will need manual testing for now. 
- Centers `AS` more on the map
- Center `HI` more on the map
- the initial map position(z/lat/long) of the US is when we click on the `48`

## How to test
Click on the territories:
1. with various zoom levels, 
2. when various types of tracts (urban and rural) are selected or not selected
3. after using search
4. after selecting a tract after searching
5. after using the zoom controls
6. anything else you can think of!

Thx!
closes #1056

### CODE REVIEW

**Review by esfoobar-usds — APPROVED** [review]
I can confirm this fixes the issue.

🍏

File: client/src/components/territoryFocusControl.tsx [hunk 1]
```diff
@@ -1,54 +1,18 @@
...
-import {LngLatBoundsLike} from 'mapbox-gl';
+import {MapEvent} from 'react-map-gl';
 
 import * as styles from './territoryFocusControl.module.scss';
 import * as EXPLORE_COPY from '../data/copy/explore';
-import * as constants from '../data/constants';
 
 interface ITerritoryFocusControl {
-  goToPlace(bounds: LngLatBoundsLike): void;
+  onClick(event: MapEvent | React.MouseEvent<HTMLButtonElement>):void;
```
Comment (saran-ahluwalia) [inline client/src/components/territoryFocusControl.tsx:9]: Slick 👍

**Comment by vim-usds** [comment]
> I can confirm this fixes the issue.
> 
> 🍏

Thanks @esfoobar-usds !!

File: client/src/components/territoryFocusControl.tsx (hunk 1 above)
Comment (vim-usds) [reply client/src/components/territoryFocusControl.tsx:9]: Thx! This is still kinda head scratcher as to why the IDE complains about the `feature` property not being in the event payload. I have to look into this. Not a show stopper but more of a personal preference to not have any linting warnings.

**Pull request merged**